onnxruntime/contrib_ops/cpu/reverse.cc

```diff
@@ -0,0 +1,244 @@
+  } else if (dtype == DataTypeImpl::GetType<BFloat16>()) {
+    ReverseImplBFloat16Type<rank>(ctx, input_tensor, output_tensor, shape, reverse_axes);
+  } else {
+    ORT_THROW("Unsupported input datatype for Reverse operator. Got ", dtype);
+  }
+}
+
+Status Reverse::Compute(OpKernelContext* p_op_kernel_context) const {
+  const auto* input = p_op_kernel_context->Input<Tensor>(0);
+  const auto& shape = input->Shape();
+  const auto& rank = shape.NumDimensions();
+  auto* output = p_op_kernel_context->Output(0, shape);
+  const auto& dtype = input->DataType();
+
+  if (rank == 0)
+    return handle_scalar_tensor(input, output, dtype);
+
+  // non-default axes - validate them for rank related correctness
+  std::vector<int64_t> copy_attr_axes_ = attr_axes_;
+  std::unordered_set<int64_t> axes_elements;
+  if (copy_attr_axes_.size() > 0) {
+    if (copy_attr_axes_.size() > rank)
+      ORT_THROW("Number of elements in axes attribute exceeds the input tensor rank in Reverse operator");
+    for (const auto& axis : copy_attr_axes_) {
+      if (axis >= static_cast<int64_t>(rank) || axis < static_cast<int64_t>(-rank))
+        ORT_THROW("Elements in axes attribute are outside bounds of the input tensor's rank in Reverse operator");
+      // check for implicit and explicit dupes in axes attribute
+      // explicit dupes (e.g.) 0 and 0
+      // implicit dupes (e.g.) -1 and 1 for an input tensor of rank 2
+      if (axis < 0) {
+        if (axes_elements.find(axis + rank) != axes_elements.end())
+          ORT_THROW("axes attribute has duplicates in Reverse operator");
+        axes_elements.insert(axis + rank);
+      } else {
+        if (axes_elements.find(axis) != axes_elements.end())
+          ORT_THROW("axes attribute has duplicates in Reverse operator");
+        axes_elements.insert(axis);
+      }
+    }
+  }
+
+  // process the output tensor's buffer
+  switch (rank) {
+    case 1:
+      ReverseImpl<1>(p_op_kernel_context, input, output, shape, copy_attr_axes_);
+      break;
+    case 2:
+      ReverseImpl<2>(p_op_kernel_context, input, output, shape, copy_attr_axes_);
+      break;
+    case 3:
+      ReverseImpl<3>(p_op_kernel_context, input, output, shape, copy_attr_axes_);
+      break;
+    case 4:
+      ReverseImpl<4>(p_op_kernel_context, input, output, shape, copy_attr_axes_);
+      break;
+    case 5:
+      ReverseImpl<5>(p_op_kernel_context, input, output, shape, copy_attr_axes_);
+      break;
+    case 6:
+      ReverseImpl<6>(p_op_kernel_context, input, output, shape, copy_attr_axes_);
+      break;
+    case 7:
+      ReverseImpl<7>(p_op_kernel_context, input, output, shape, copy_attr_axes_);
+      break;
+    case 8:
+      ReverseImpl<8>(p_op_kernel_context, input, output, shape, copy_attr_axes_);
+      break;
+    default:
+      ORT_THROW("Reverse operator is not implemented for input tensors with 9 or more dimensions (rank)");
+  }
```

Comment:
Could include the value of 'rank' in the error message.

Reply:
Added this to the error message